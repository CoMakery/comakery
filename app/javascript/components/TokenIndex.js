import React from "react"
import PropTypes from "prop-types"

class TokenIndex extends React.Component {
  render () {
    return (
      <React.Fragment>
        {this.props.tokens.map((t) =>
          <p key={t.name}>{t.name} â€“ {t.id}</p>
        )}
      </React.Fragment>
    );
  }
}

TokenIndex.propTypes = {
  tokens   : PropTypes.array.isRequired
}
TokenIndex.defaultProps = {
  tokens   : []
}
export default TokenIndex
