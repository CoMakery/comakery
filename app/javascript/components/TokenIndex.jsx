import React from 'react'
import PropTypes from 'prop-types'
import Layout from './layouts/Layout'
import SidebarItem from './styleguide/SidebarItem'
import SidebarItemBold from './styleguide/SidebarItemBold'

class TokenIndex extends React.Component {
  constructor(props) {
    super(props)
    this.handleListClick = this.handleListClick.bind(this)
    this.state = {
      selectedToken: null
    }
  }

  handleListClick(token) {
    this.setState({
      selectedToken: token
    })
  }

  render() {
    return (
      <React.Fragment>
        <Layout
          className="token-index"
          title="Tokens"
          sidebar={
            <React.Fragment>
              <div className="token-index--sidebar">
                <SidebarItemBold
                  className="token-index--sidebar--item__bold"
                  iconLeftName="MARK-WHITE.svg"
                  iconRightName="PLUS.svg"
                  text="Create a Token"
                  onClick={(_) => window.location = '/tokens/new'}
                />

                <hr />

                <div className="token-index--sidebar--info">
                  Please select token:
                </div>

                {this.props.tokens.map((t) =>
                  <SidebarItem
                    className="token-index--sidebar--item"
                    key={t.id}
                    iconLeftUrl={t.logoUrl}
                    iconRightName="REARRANGE.svg"
                    text={`${t.name} (${t.symbol})`}
                    selected={this.state.selectedToken === t}
                    onClick={(_) => this.handleListClick(t)}
                  />
                )}
              </div>
            </React.Fragment>
          }
        >
          {this.state.selectedToken &&
            <div className="token-index--view">
              <div className="token-index--view--logo">
                <img
                  src={this.state.selectedToken.logoUrl}
                />
              </div>

              <div className="token-index--view--info">
                <div className="token-index--view--info--item">
                  <div className="token-index--view--info--item--name">
                    token name
                  </div>
                  <div className="token-index--view--info--item--value">
                    {this.state.selectedToken.name}
                  </div>
                </div>

                <div className="token-index--view--info--item">
                  <div className="token-index--view--info--item--name">
                    payment type
                  </div>
                  <div className="token-index--view--info--item--value">
                    {this.state.selectedToken.coinType}
                  </div>
                </div>

                <div className="token-index--view--info--item">
                  <div className="token-index--view--info--item--name">
                    token symbol
                  </div>
                  <div className="token-index--view--info--item--value">
                    {this.state.selectedToken.symbol}
                  </div>
                </div>

                { this.state.selectedToken.contractAddress &&
                  <div className="token-index--view--info--item">
                    <div className="token-index--view--info--item--name">
                      contract address
                    </div>
                    <div className="token-index--view--info--item--value">
                      {this.state.selectedToken.contractAddress}
                    </div>
                  </div>
                }

                { this.state.selectedToken.ethereumContractAddress &&
                  <div className="token-index--view--info--item">
                    <div className="token-index--view--info--item--name">
                      contract address
                    </div>
                    <div className="token-index--view--info--item--value">
                      {this.state.selectedToken.ethereumContractAddress}
                    </div>
                  </div>
                }

                { this.state.selectedToken.blockchainNetwork &&
                  <div className="token-index--view--info--item">
                    <div className="token-index--view--info--item--name">
                      blockchain network
                    </div>
                    <div className="token-index--view--info--item--value">
                      {this.state.selectedToken.blockchainNetwork}
                    </div>
                  </div>
                }

                { this.state.selectedToken.ethereumNetwork &&
                  <div className="token-index--view--info--item">
                    <div className="token-index--view--info--item--name">
                      blockchain network
                    </div>
                    <div className="token-index--view--info--item--value">
                      {this.state.selectedToken.ethereumNetwork}
                    </div>
                  </div>
                }
              </div>

              <div className="token-index--view--link">
                <a
                  href={`/tokens/${this.state.selectedToken.id}`}
                >
                  edit token
                </a>
              </div>
            </div>
          }
        </Layout>
      </React.Fragment>
    )
  }
}

TokenIndex.propTypes = {
  tokens: PropTypes.array.isRequired
}
TokenIndex.defaultProps = {
  tokens: []
}
export default TokenIndex
