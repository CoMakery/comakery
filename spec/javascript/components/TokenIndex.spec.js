import React from 'react'
import { mount } from 'enzyme'
import TokenIndex from 'components/TokenIndex'

describe('TokenIndex', () => {
  it('renders correctly without props', () => {
    const wrapper = mount(<TokenIndex />)

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.exists('.token-index')).toBe(true)
    expect(wrapper.exists('.token-index--sidebar')).toBe(true)
    expect(wrapper.exists('.token-index--sidebar SidebarItemBold')).toBe(true)
    expect(wrapper.find('.token-index--sidebar SidebarItemBold').props().iconLeftName).toBe('MARK-WHITE.svg')
    expect(wrapper.find('.token-index--sidebar SidebarItemBold').props().iconRightName).toBe('PLUS.svg')
    expect(wrapper.exists('.token-index--sidebar--info')).toBe(false)
    expect(wrapper.exists('.token-index--view')).toBe(false)
  })

  it('renders correctly with tokens', () => {
    const tokens = [
      {
        'id'                     : 0,
        'name'                   : 'ERC-TEST',
        '_tokenType'               : 'erc20',
        '_blockchain'      : 'main',
        'contractAddress': '0x00',
        'symbol'                 : 'ERCT',
        'decimalPlaces'          : 2,
        'logoUrl'                : '/ERCT.png'
      },
      {
        'id'                     : 1,
        'name'                   : 'QRC-TEST',
        '_tokenType'               : 'qrc20',
        '_blockchain'      : 'test',
        'contractAddress'        : '0',
        'symbol'                 : 'QRCT',
        'decimalPlaces'          : 0,
        'logoUrl'                : '/QRCT.png'
      }
    ]
    const wrapper = mount(<TokenIndex tokens={tokens} />)

    expect(wrapper.find('.token-index--sidebar--info').text()).toBe('Please select token:')

    expect(wrapper.exists('SidebarItem[iconLeftUrl="/ERCT.png"]')).toBe(true)
    expect(wrapper.exists('SidebarItem[text="ERC-TEST (ERCT)"]')).toBe(true)

    expect(wrapper.exists('SidebarItem[iconLeftUrl="/QRCT.png"]')).toBe(true)
    expect(wrapper.exists('SidebarItem[text="QRC-TEST (QRCT)"]')).toBe(true)
  })

  it('displays correct token details on sidebar item click', () => {
    const tokens = [
      {
        'id'                     : 0,
        'name'                   : 'ERC-TEST',
        '_tokenType'               : 'erc20',
        '_blockchain'      : 'main',
        'contractAddress': '0x00',
        'symbol'                 : 'ERCT',
        'decimalPlaces'          : 2,
        'logoUrl'                : '/ERCT.png'
      },
      {
        'id'                     : 1,
        'name'                   : 'QRC-TEST',
        '_tokenType'               : 'qrc20',
        '_blockchain'      : 'test',
        'contractAddress'        : 0,
        'symbol'                 : 'QRCT',
        'decimalPlaces'          : 0,
        'logoUrl'                : '/QRCT.png'
      }
    ]
    const wrapper = mount(<TokenIndex tokens={tokens} />)

    expect(wrapper.exists('.token-index--view')).toBe(false)

    wrapper.find('SidebarItem[iconLeftUrl="/ERCT.png"]').simulate('click')
    expect(wrapper.find('.token-index--view').text()).toMatch(/ERC-TEST/)

    wrapper.find('SidebarItem[iconLeftUrl="/QRCT.png"]').simulate('click')
    expect(wrapper.find('.token-index--view').text()).not.toMatch(/ERC-TEST/)
    expect(wrapper.find('.token-index--view').text()).toMatch(/QRC-TEST/)
  })
})
