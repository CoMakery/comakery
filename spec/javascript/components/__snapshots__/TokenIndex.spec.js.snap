// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TokenIndex renders correctly without props 1`] = `
<TokenIndex
  paginationHtml=""
  tokens={Array []}
>
  <Layout
    category=""
    className="token-index"
    customTitle={null}
    hasBackButton={false}
    sidebar={
      <React.Fragment>
        <div
          className="token-index--sidebar"
        >
          <a
            href="/tokens/new"
          >
            <SidebarItemBold
              className="token-index--sidebar--item__bold"
              iconLeftName="MARK-WHITE.svg"
              iconRightName="PLUS.svg"
              text="Create a Token"
            />
          </a>
          <ForwardRef(styled.div)
            dangerouslySetInnerHTML={
              Object {
                "__html": "",
              }
            }
          />
        </div>
      </React.Fragment>
    }
    subfooter={null}
    title="Tokens"
  >
    <div
      className="layout token-index"
    >
      <div
        className="layout--content"
      >
        <div
          className="layout--content--title"
        >
          Tokens
        </div>
        <div
          className="layout--content--wrapper"
        >
          <div
            className="layout--content--sidebar"
          >
            <div
              className="token-index--sidebar"
            >
              <a
                href="/tokens/new"
              >
                <SidebarItemBold
                  className="token-index--sidebar--item__bold"
                  iconLeftName="MARK-WHITE.svg"
                  iconRightName="PLUS.svg"
                  text="Create a Token"
                >
                  <SidebarItem
                    className="sidebar-item__bold token-index--sidebar--item__bold"
                    iconLeftName="MARK-WHITE.svg"
                    iconLeftUrl=""
                    iconRightName="PLUS.svg"
                    notificationColor=""
                    selected={false}
                    text="Create a Token"
                  >
                    <div
                      className="sidebar-item--wrapper"
                    >
                      <div
                        className="sidebar-item sidebar-item__bold token-index--sidebar--item__bold"
                      >
                        <span
                          className="sidebar-item--icon-left"
                        >
                          <Icon
                            className=""
                            name="MARK-WHITE.svg"
                          >
                            <img
                              className="icon-legacy icon__MARK-WHITE-svg"
                              src="test-file-stub"
                            />
                          </Icon>
                        </span>
                        <div
                          className="sidebar-item--content"
                        >
                          <div
                            className="sidebar-item--text"
                          >
                            Create a Token
                          </div>
                          <div
                            className="sidebar-item--subchild"
                          />
                        </div>
                        <span
                          className="sidebar-item--icon-right"
                        >
                          <Icon
                            className=""
                            name="PLUS.svg"
                          >
                            <img
                              className="icon-legacy icon__PLUS-svg"
                              src="test-file-stub"
                            />
                          </Icon>
                        </span>
                        <div
                          className="sidebar-item--left-border"
                        />
                        <div
                          className="sidebar-item--right-border sidebar-item--right-border__"
                        />
                      </div>
                    </div>
                  </SidebarItem>
                </SidebarItemBold>
              </a>
              <styled.div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "",
                  }
                }
              >
                <StyledComponent
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "",
                    }
                  }
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-bdVaJa",
                        "isStatic": false,
                        "lastClassName": "eDrlXv",
                        "rules": Array [
                          "
  text-align: right;
  width: 100%;
  padding: 15px 0;
",
                        ],
                      },
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-bdVaJa",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="sc-bdVaJa eDrlXv"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "",
                      }
                    }
                  />
                </StyledComponent>
              </styled.div>
            </div>
          </div>
          <div
            className="layout--content--content__sidebared animated fadeIn faster"
          />
        </div>
      </div>
    </div>
  </Layout>
</TokenIndex>
`;
